tests/rptest/tests/topic_delete_test.py

```diff
@@ -87,30 +87,115 @@ def __init__(self, test_context):
             num_brokers=test_context.cluster.available().size(),
             si_settings=self.si_settings)
 
+        self._s3_port = self.si_settings.cloud_storage_api_endpoint_port
+
         self.kafka_tools = KafkaCliTools(self.redpanda)
 
-    def _populate_topic(self):
+    def _populate_topic(self, topic_name):
         """
         Get system into state where there is data in both local
         and remote storage for the topic.
         """
         # Set retention to 5MB
         self.kafka_tools.alter_topic_config(
-            self.topic, {'retention.local.target.bytes': 5 * 1024 * 1024})
+            topic_name, {'retention.local.target.bytes': 5 * 1024 * 1024})
 
         # Write out 10MB
-        self.kafka_tools.produce(self.topic,
+        self.kafka_tools.produce(topic_name,
                                  record_size=4096,
                                  num_records=2560)
 
         # Wait for segments evicted from local storage
         for i in range(0, 3):
-            wait_for_segments_removal(self.redpanda, self.topic, i, 5)
+            wait_for_segments_removal(self.redpanda, topic_name, i, 5)
 
         # Confirm objects in remote storage
-        before_objects = self.s3_client.list_objects(
-            self.si_settings.cloud_storage_bucket)
-        assert sum(1 for _ in before_objects) > 0
+        objects = self.s3_client.list_objects(
+            self.si_settings.cloud_storage_bucket, topic=topic_name)
+        assert sum(1 for _ in objects) > 0
+
+    @cluster(
+        num_nodes=3,
+        log_allow_list=[
+            'exception while executing partition operation: {type: deletion'
+        ])
+    def topic_delete_unavailable_test(self):
+        """
+        Test deleting while the S3 backend is unavailable: we should see
+        that local deletion proceeds, and remote deletion eventually
+        gives up.
+        """
+        self._populate_topic(self.topic)
+        objects_before = set(
+            self.redpanda.s3_client.list_objects(
+                self.si_settings.cloud_storage_bucket, topic=self.topic))
+        assert (len(objects_before) > 0)
```

Comment:
nit: Extra set of parens around logical condition

Reply:
cleaned this up while rebasing